File: scripts/test-scrape.py
#!/usr/bin/env python3
"""
Test script for scraping API (Python version)
Usage: python3 scripts/test-scrape.py <url> [modules...]
Example: python3 scripts/test-scrape.py "https://www2.hm.com/en_us/men/sale/view-all.html" products images
"""

import sys
import json
import requests
import time
from typing import List, Dict, Any

API_URL = "http://localhost:3000"

def test_scrape(url: str, modules: List[str]) -> None:
    """Test the scrape API endpoint"""
    print(f"\n🧪 Testing scrape: {url}")
    print(f"📦 Modules: {', '.join(modules)}\n")
    
    try:
        start_time = time.time()
        response = requests.post(
            f"{API_URL}/api/scrape",
            json={
                "url": url,
                "modules": modules,
            },
            headers={"Content-Type": "application/json"},
            timeout=60,
        )
        
        duration = int((time.time() - start_time) * 1000)
        
        if not response.ok:
            print(f"❌ Error: {response.status_code} {response.text[:200]}")
            sys.exit(1)
        
        data = response.json()
        
        if not data.get("success"):
            print(f"❌ Scrape failed: {data.get('error', 'Unknown error')}")
            sys.exit(1)
        
        print(f"✅ Success ({duration}ms)\n")
        
        results = data.get("data", {})
        
        # Products
        if "products" in results:
            products = results["products"]
            print(f"📦 Products: {len(products)}")
            if products:
                sample = products[0]
                print("   Sample:", {
                    "title": (sample.get("title") or "")[:50] or "N/A",
                    "price": sample.get("price") or "N/A",
                    "hasImage": bool(sample.get("image")),
                    "hasLink": bool(sample.get("link")),
                })
        
        # Images
        if "images" in results:
            print(f"🖼️  Images: {len(results['images'])}")
        
        # Contacts
        if "contacts" in results:
            contacts = results["contacts"]
            emails = contacts.get("emails", [])
            phones = contacts.get("phones", [])
            socials = contacts.get("socials", [])
            print(f"📧 Contacts: {len(emails)} emails, {len(phones)} phones, {len(socials)} socials")
        
        # Text
        if "text" in results:
            text = results["text"]
            headings = text.get("headings", [])
            paragraphs = text.get("paragraphs", [])
            print(f"📝 Text: {len(headings)} headings, {len(paragraphs)} paragraphs")
        
        # Assets
        if "assets" in results:
            print(f"📎 Assets: {len(results['assets'])}")
        
        # Crawl
        if "crawl" in results:
            print(f"🔗 Links: {len(results['crawl'])}")
        
        print("\n")
        
    except requests.exceptions.ConnectionError:
        print("❌ Connection error: Could not connect to API")
        print("\nMake sure the dev server is running: npm run dev")
        sys.exit(1)
    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python3 scripts/test-scrape.py <url> [modules...]")
        sys.exit(1)
    
    url = sys.argv[1]
    modules = sys.argv[2:] if len(sys.argv) > 2 else ["products"]
    
    test_scrape(url, modules)

File: scripts/test-scrape.mjs
#!/usr/bin/env node

/**
 * Test script for scraping API
 * Usage: node scripts/test-scrape.mjs <url> [modules...]
 * Example: node scripts/test-scrape.mjs "https://www2.hm.com/en_us/men/sale/view-all.html" products images
 */

const url = process.argv[2];
const modules = process.argv.slice(3).length > 0 ? process.argv.slice(3) : ['products'];

if (!url) {
  console.error('Usage: node scripts/test-scrape.mjs <url> [modules...]');
  process.exit(1);
}

const ENV_API_URL = process.env.API_URL || process.env.SCRAPE_API_URL;
const API_CANDIDATES = ENV_API_URL
  ? [ENV_API_URL]
  : [
      'http://127.0.0.1:3001',
      'http://localhost:3001',
      'http://127.0.0.1:3000',
      'http://localhost:3000',
    ];

async function requestScrape(apiBase) {
  return fetch(`${apiBase}/api/scrape`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      url,
      modules,
    }),
  });
}

async function testScrape() {
  console.log(`\n🧪 Testing scrape: ${url}`);
  console.log(`📦 Modules: ${modules.join(', ')}\n`);

  try {
    const startTime = Date.now();
    let response = null;
    let apiBase = '';
    let lastNetworkError = null;

    for (const candidate of API_CANDIDATES) {
      try {
        const candidateResponse = await requestScrape(candidate);
        response = candidateResponse;
        apiBase = candidate;
        break;
      } catch (error) {
        lastNetworkError = error;
      }
    }

    if (!response) {
      throw lastNetworkError || new Error('No reachable local scrape API endpoint found');
    }

    const duration = Date.now() - startTime;
    const data = await response.json();

    if (!response.ok) {
      console.log(`🔌 API: ${apiBase}`);
      console.error('❌ Error:', data.error || response.statusText);
      process.exit(1);
    }

    if (!data.success) {
      console.error('❌ Scrape failed:', data.error);
      process.exit(1);
    }

    console.log(`🔌 API: ${apiBase}`);
    console.log(`✅ Success (${duration}ms)\n`);

    const results = data.data || {};

    // Products
    if (results.products) {
      console.log(`📦 Products: ${results.products.length}`);
      if (results.products.length > 0) {
        const sample = results.products[0];
        console.log('   Sample:', {
          title: sample.title?.substring(0, 50) || 'N/A',
          price: sample.price || 'N/A',
          hasImage: !!sample.image,
          hasLink: !!sample.link,
        });
      }
    }

    // Images
    if (results.images) {
      console.log(`🖼️  Images: ${results.images.length}`);
    }

    // Videos
    if (results.videos) {
      console.log(`🎬 Videos: ${results.videos.length}`);
    }

    // Contacts
    if (results.contacts) {
      const { emails, phones, socials } = results.contacts;
      console.log(`📧 Contacts: ${emails?.length || 0} emails, ${phones?.length || 0} phones, ${socials?.length || 0} socials`);
    }

    // Text
    if (results.text) {
      console.log(`📝 Text: ${results.text.headings?.length || 0} headings, ${results.text.paragraphs?.length || 0} paragraphs`);
    }

    // Assets
    if (results.assets) {
      console.log(`📎 Assets: ${results.assets.length}`);
    }

    // Crawl
    if (results.crawl) {
      console.log(`🔗 Links: ${results.crawl.length}`);
    }

    console.log('\n');
  } catch (error) {
    console.error('❌ Network error:', error.message);
    console.error('\nMake sure the dev server is running (port 3001 or 3000), or set API_URL.');
    process.exit(1);
  }
}

testScrape();
